Add example with parameters

tomanagle opened the pull request:
- Replaces the PI example with an example that uses parameters to show that the key must be unique for each resource
- Adds an extra line of explainer to the key option in the readme and common.ts

Xiphe commented on readme.md:
````diff
@@ -69,12 +69,36 @@ console.log(await getPi());
 // that now replaces current cache entry and is cached for 5 minutes
 ```
````
I'd get rid of this example in favor of yours

Xiphe commented on readme.md:
```diff
@@ -69,12 +69,36 @@ console.log(await getPi());
+    key: `users_${userId}`,
+    cache: lru,
+    async getFreshValue() {
+     const response fetch(`https://example.com/users/${userId}`)
```
theres an `=` missing and indentation is wrong

Xiphe commented on readme.md:
```diff
@@ -69,12 +69,36 @@ console.log(await getPi());
+    cache: lru,
+    async getFreshValue() {
+     const response fetch(`https://example.com/users/${userId}`)
+    return response.json();
```
please add two more spaces for indentation

Xiphe reviewed (changes requested):
Awesome! Thanks for the contribution and input. I have some minor changes I'd like to see with this. If you want I can also add them, just let me know.

tomanagle commented:
Nah, all good. I'll update the PR now. Thanks for the feedback

tomanagle commented:
I've updated the PR. I removed the PI example and added the `getUserById` example. I also changed the URL to something that will work, just because people may want to copy it and run it to see t working.

Xiphe commented:
awesome! thank you for the contribution 🎉
